packages/cache/src/cache.ts

```diff
@@ -182,17 +172,41 @@ export async function saveCache(
       await listTar(archivePath, compressionMethod)
     }
 
-    const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
-      throw new Error(
-        `Cache size of ~${Math.round(
-          archiveFileSize / (1024 * 1024)
-        )} MB (${archiveFileSize} B) is over the 10GB limit, not saving cache.`
```

Comment:
Don't remove this. We may need it for backward compatiblity. You can put a check so that it only runs for non-GHES.

Reply:
> Can we not send size even for non-GHES? There is no harm and server side can decide whether to process it or not (which currently it will not do). But sending every-time just simplifies the logic a bit here.

@bishal-pdMSFT 
If we wish to send `cacheSize` for all the cases wont it be much cleaner to not check anything here?
We have updated the tests accordingly so what else will require backward compatibility as the behaviour for user is still same?

Reply:
Backward compatibility in the sense that the action/toolkit should still do the checking even it someone hits an older service which does not have the service side checking. There are two cases in which such need arises:
1. if the service side change in rolled back for some reason (lets say a critical bug on the service side). Since the client is shipped we can't hide/un-ship it, such clients will not have this check happening. Also applies if service side change in only rolled out to few rings only.
2. If the action is run against an older GHES server (<3.5) then that server will not have server side check. This is actually not a problem for us as cache service does not exist in older GHES versions but I guess the principal for compatibility can still apply.

TBH these are not big reasons in our case but I guess its good to have backward compatibility as a principle.